src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -25,9 +39,111 @@ public FindCommand parse(String args) throws ParseException {
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the user input does not conform to {@code Phone} format.
+     */
+    public String parsePhone(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_PHONE);
+        if (!arePrefixesPresent(argMultimap, PREFIX_PHONE)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        }
+        Phone phone = ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).get());
+        return phone.value;
+    }
 
-        return new FindCommand(new NameContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
+    /**
+     * Parses the given {@code String} of arguments into a {@code String email}
+     * which will be used for {@code FindCommandParser} object for execution.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the user input does not conform to {@code Email} format.
+     */
+    public String parseEmail(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_EMAIL);
+        if (!arePrefixesPresent(argMultimap, PREFIX_EMAIL)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        }
+        Email email = ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get());
+        return email.value;
+    }
+
+    /**
+     * Parses the given {@code String} of arguments into a {@code String academicMajor}
+     * which will be used for {@code FindCommandParser} object for execution.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the user input does not conform to {@code AcademicMajor} format.
+     */
+    public String parseAcademicMajor(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_ACADEMIC_MAJOR);
+        if (!arePrefixesPresent(argMultimap, PREFIX_ACADEMIC_MAJOR)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        }
+        AcademicMajor academicMajor = ParserUtil.parseAcademicMajor(argMultimap.getValue(PREFIX_ACADEMIC_MAJOR).get());
+        return academicMajor.value;
     }
 
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
```

Comment:
In the case where we decide to remove all the above methods, this will also not be required.

Reply:
yeap okay let me check whether it works by removing and adding toString()